aws-s3-transfer-manager/src/operation/download/discovery.rs

```diff
@@ -108,67 +97,33 @@ pub(super) async fn discover_obj(
     Ok(discovery)
 }
 
-async fn discover_obj_with_head(
-    ctx: &DownloadContext,
-    input: &DownloadInput,
-    byte_range: Option<ByteRange>,
-) -> Result<ObjectDiscovery, crate::error::Error> {
-    let resp = ctx
-        .client()
-        .head_object()
-        .set_bucket(input.bucket().map(str::to_string))
-        .set_key(input.key().map(str::to_string))
-        .send()
-        .await
-        .map_err(error::discovery_failed)?;
-    let object_meta: ObjectMetadata = resp.into();
-
-    let remaining = match byte_range {
-        Some(range) => match range {
-            ByteRange::Inclusive(start, end) => start..=end,
-            ByteRange::AllFrom(start) => start..=object_meta.content_length(),
-            ByteRange::Last(n) => {
-                (object_meta.content_length() - n + 1)..=object_meta.content_length()
-            }
-        },
-        None => 0..=object_meta.content_length(),
-    };
-
-    Ok(ObjectDiscovery {
-        remaining,
-        chunk_meta: None,
-        object_meta,
-        initial_chunk: None,
-    })
-}
-
-async fn discover_obj_with_get(
-    ctx: &DownloadContext,
-    input: GetObjectInputBuilder,
+// Use 1..=0 to represent an empty object, and clippy doesn't allow that.
+#[allow(clippy::reversed_empty_ranges)]
```

Comment:
You might be able to avoid this all together by creating a const:

```rs
const EMPTY_RANGE: RangeInclusive<u64> = RangeInclusive::new(0, 0);

```